dbms/src/Storages/S3/FileCache.cpp

```
@@ -980,6 +985,43 @@ bool FileCache::finalizeReservedSize(FileType reserve_for, UInt64 reserved_size,
+    PosixWritableFile ofile(fname, true, O_CREAT | O_WRONLY, 0666, write_limiter);
+    // simply create an empty file
+    if (unlikely(content_length <= 0))
+        return 0;
+
+    GET_METRIC(tiflash_storage_remote_cache_bytes, type_dtfile_download_bytes).Increment(content_length);
+    size_t total_written = 0;
+    static const Int64 MAX_BUFFER_SIZE = 16 * 1024; // 16k
+    ReadBufferFromIStream rbuf(istr, std::min(content_length, MAX_BUFFER_SIZE));
+    while (!rbuf.eof())
+    {
+        size_t count = rbuf.buffer().end() - rbuf.position();
```

Clang (14.0.6)

[error] variable 'count' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)